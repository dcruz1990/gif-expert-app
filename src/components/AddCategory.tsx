import React, { useState } from "react";

type categoryProps = {
  setCategories: Function;
};

export const AddCategory = ({ setCategories }: categoryProps) => {
  const [inputvalue, setInputValue] = useState<string>("");

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    if (inputvalue.length === 0) {
      return false;
    } else {
      setCategories(inputvalue);
      setInputValue("");
    }
  };

  return (
    <>
      <h2>Add Category</h2>
      <form action="" onSubmit={handleSubmit}>
        <input
          value={inputvalue}
          id="category"
          type="text"
          onChange={handleInputChange}
        />
        <button disabled={inputvalue.length === 0} type="submit">
          Add Category
        </button>
      </form>
    </>
  );
};
