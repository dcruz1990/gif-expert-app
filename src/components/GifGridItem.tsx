import React from "react";

interface props {
  url: string;
  title: string;
}

export const GifGridItem = ({ url, title }: props) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};
