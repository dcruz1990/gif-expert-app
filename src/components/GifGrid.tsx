import React, { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";
import { GifGridItem } from "./GifGridItem";

type gridProps = {
  category: string;
};

export const GifGrid = ({ category }: gridProps) => {
  const [images, setImages] = useState<
    {
      id: string;
      url: string;
      title: string;
    }[]
  >();
  // onMounted de Vue
  useEffect(() => {
    getGifs(category).then((data) => {
      setImages(data);
    });
  }, [category]);

  return (
    <div>
      <h3>{category}</h3>

      <div className="cardContainer">
        {images?.map((image) => {
          return (
            <GifGridItem key={image.id} url={image.url} title={image.title} />
          );
        })}
      </div>
    </div>
  );
};
