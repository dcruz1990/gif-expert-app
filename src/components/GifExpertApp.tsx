import React, { useState } from "react";
import { AddCategory } from "./AddCategory";
import { GifGrid } from "./GifGrid";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Naruto"]);

  const handleAdd = (value: string) => {
    setCategories([value, ...categories]);
  };

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />

      <AddCategory setCategories={handleAdd} />

      <ol>
        {categories.map((item) => (
          <GifGrid category={item} key={item} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
