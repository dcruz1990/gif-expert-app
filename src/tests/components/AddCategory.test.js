import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe('Pruebas en AddCategory', () => {

    const setCategories = () => {}
    test('should show correctly', () => {
        const wrapper = shallow(<AddCategory setCategories={AddCategory}/>)
        expect(wrapper).toMatchSnapshot()
    })
    
})