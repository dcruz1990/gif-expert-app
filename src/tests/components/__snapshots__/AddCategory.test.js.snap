// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory should show correctly 1`] = `
<Fragment>
  <h2>
    Add Category
  </h2>
  <form
    action=""
    onSubmit={[Function]}
  >
    <input
      id="category"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      disabled={true}
      type="submit"
    >
      Add Category
    </button>
  </form>
</Fragment>
`;
